File: agent-driven-architecture-main/examples/typescript/src/blocks/analysis/port.ts
// ── blocks/analysis/port — the relay's WRITE half (11.2) ─────────────────────
// An interface, nothing else (check C11). The block owns its dependency; the
// composition root binds it, exactly like `OncallPort` and `DeliveryPort`.
//
// The relay is SPLIT BY DIRECTION, and the split is the architecture's own line:
// the READ side is `spine/ports/relay` because the consumer must bound it and
// the `Recall` it produces is spine transport that rides `StepRecord`; the WRITE
// side is a feature's backend seam and lives here. `app/wire` binds both halves
// to one store, because the root is the only cross-layer importer — which is how
// the two tiers meet without the spine ever naming a block.

import type { Timestamp } from "@adr/spine/pure/ids";

export interface AnalysisRelay {
  publish(at: Timestamp, text: string): void;
}

File: agent-driven-architecture-main/examples/kotlin/block/analysis/src/main/kotlin/adr/blocks/analysis/Port.kt
// ── blocks/analysis/port — the WRITE half of the relay (4.6/G11) ───────────
// The relay is split BY DIRECTION, and that split is the architecture's own line:
//
//   READ   spine/ports/relay      the fast tier must BOUND it, and the `Recall` it
//                                 produces is spine transport riding StepRecord.
//   WRITE  here                   a feature's dependency, bound at the root — exactly
//                                 like OncallPort and DeliveryPort.
//
// An interface only. The concrete relay lives in spine/concurrency/in-memory and
// app/wire adapts it to this port, because the root is the only cross-layer
// importer (G7/G10). The spine therefore still names no block (gate check C15).

package adr.blocks.analysis

import adr.spine.pure.Timestamp

interface AnalysisRelay {
    /** Append one conclusion. `at` is the boundary's clock read, handed in (G9). */
    fun publish(at: Timestamp, text: String)
}
